java/ql/src/Security/CWE/CWE-927/ImplicitPendingIntents.qhelp

```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+
+<qhelp>
+
+<overview>
+<p>A <code>PendingIntent</code> describes an action in the form of an Intent that is intended to be given and executed 
+  at a later time by another application. The Intent wrapped by a <code>PendingIntent</code> is executed on behalf of 
+  the application that created it, and with its same privileges.</p>
+<p>If a <code>PendingIntent</code> is configured to be mutable, the fields of its internal Intent can be changed by the
+  receiving application if they were not previously set. This means that a mutable <code>PendingIntent</code> that has 
+  not defined a destination component (that is, an implicit <code>PendingIntent</code>) can be altered to execute an 
+  arbitrary action with the privileges of the application that created it.</p>
+<p>If an implicit <code>PendingIntent</code> is obtainable by a malicious application by any of the following means:</p>
+<ul>
+  <li>It is wrapped and sent as an extra of another implicit Intent</li>
+  <li>It is sent as the action of a Slide</li>
+  <li>It is sent as the action of a Notification</li>
+</ul>
+<p></p>
+<p>the attacker could modify the underlying Intent and execute an arbitrary action with elevated privileges. 
+  This could give the malicious application access to private components of the victim application, 
+  or the ability to perform actions without having the necessary permissions.</p>
+</overview>
+
+<recommendation>
+<p>Avoid creating implicit <code>PendingIntent</code>s. This means that the underlying Intent should always have an 
+  explicit destination component.</p>
+<p>Also, when adding the <code>PendingIntent</code> as an extra of another Intent, make sure that said Intent also has
+an explicit destination component, so that it is not delivered to untrusted applications.</p>
+<p>It is also recommended to create the <code>PendingIntent</code> using the flag <code>FLAG_IMMUTABLE</code> whenever 
+  possible, to prevent the destination component from modifying empty fields of the underlying Intent.</p>
```

Comment:
Suggest simplifying to:

>Create the <code>PendingIntent</code> using the flag <code>FLAG_IMMUTABLE</code> whenever 
  possible, to prevent the destination component from modifying empty fields of the underlying <code>Intent</code>.